src/Stratis.Bitcoin/P2P/Peer/INetworkPeer.cs

```diff
@@ -159,5 +159,21 @@ public interface INetworkPeer : IDisposable
         /// <typeparam name="T">Type of the behavior to find.</typeparam>
         /// <returns>Collection of behaviors of specific type.</returns>
         T Behavior<T>() where T : INetworkPeerBehavior;
+
+        /// <summary>
+        /// Determines if this peer matches the ip and optional port.
+        /// </summary>
+        /// <param name="ip">The ip to match.</param>
+        /// <param name="port">The port to match (optional).</param>
+        /// <returns><c>True</c> if the endpoint matches and <c>false</c> otherwise.</returns>
+        bool Match(IPAddress ip, int? port = null);
+
+        /// <summary>
+        /// Determines if this peer matches the endpoint. Port matching can be suppressed by passing an optional argument.
+        /// </summary>
+        /// <param name="ep">The endpoint to match.</param>
+        /// <param name="matchPort">Set to <c>false</c> to omit port matching.</param>
+        /// <returns><c>True</c> if the endpoint matches and <c>false</c> otherwise.</returns>
+        bool Match(IPEndPoint ep, bool matchPort = true);
```

Comment:
I'm just thinking that we should perhaps rename this as we already have extension called "Match" which matches `IPEndPoint`s in `Stratis.Bitcoin.Utilities.Extensions`?

Reply:
Renamed to `HasRemoteIPAdress` and `HasRemoteEndPoint`.